Start implementing validation of business rules

AntonHermann opened the pull request:
My idea on how validation of business rules could look like, with 2 example rules implemented.

Currently, the target profile isn't used, but that would be easy to add (e.g. one list of BRs for each profile)

FaultierSP commented:
I like the idea, but would the file be better placed in the "components" folder?

AntonHermann commented:
Sure, I don't have any particular opinions on that :) will update the PR tomorrow at work

AntonHermann commented:
FYI the last commit ([52072ab](https://github.com/FaultierSP/zugferd_rust/pull/7/commits/52072aba0af3910263ba8bf4adfc15c05928b139)) changes the code quite a bit. Just so that you don't overlook that. The rest didn't change, I squashed 1 fixup commit and moved everything into components/
